crates/payload/basic/src/lib.rs

```diff
@@ -514,6 +547,20 @@ struct PayloadConfig {
     attributes: PayloadBuilderAttributes,
     /// The chain spec.
     chain_spec: Arc<ChainSpec>,
+    /// The rollup's compute pending block option
+    #[cfg(feature = "optimism")]
+    compute_pending_block: bool,
```

Comment:
this makes me think we need something extra in the Payload generator so we don't need all the cfgs